src/platform.ts

```diff
@@ -48,7 +57,33 @@ export class EsphomePlatform implements DynamicPlatformPlugin {
     }
 
     protected onHomebridgeDidFinishLaunching(): void {
-        this.config.devices?.forEach((deviceConfig) => {
+        let devices: Observable<IEsphomeDeviceConfig> = from(this.config.devices ?? []);
+        if (this.config.discover) {
+            const excludeConfigDevices: Set<string> = new Set();
+            devices = concat(
+                discoverDevices(this.config.discoveryTimeout ?? DEFAULT_DISCOVERY_TIMEOUT).pipe(
```

Comment:
From my experience the discovery is super fast in any case - my test device is found in ~1/2 of a second so I doubt if performance is an issue here. From the other side setting up explicitly specified devices before running discovery will not allow to use bonjour-supplied IPv4 addresses (which feels more reliable to me) instead of mDNS hostnames. At least i had a few issues with Homebridge/operating system not being able to resolve my esphome's .local hostname to address and thus not being able to connect and the current approach helped me to workaround this issue